### PR TITLE
docs(nx-cloud): bitbucket data center docs

### PR DESCRIPTION
**Pull request by rarmatei**
- Adds docs for the new bitbucket data center integration + auth
- Refactors some of the other integration docs to be consistent with each other and link to the Helm repo where possible

### CODE REVIEW

File: nx-dev/nx-dev/redirect-rules.js [hunk 1]
```diff
@@ -418,7 +418,9 @@ const nxCloudUrls = {
   '/nx-cloud/set-up/github':
     '/nx-cloud/recipes/source-control-integration/github',
   '/nx-cloud/set-up/bitbucket-cloud':
-    '/nx-cloud/recipes/source-control-integration/bitbucket-cloud',
+    '/nx-cloud/recipes/source-control-integration/bitbucket',
+  '/nx-cloud/recipes/source-control-integration/bitbucket-cloud':
+    '/nx-cloud/recipes/source-control-integration/bitbucket',
```
Comment (ndcunningham) [inline nx-dev/nx-dev/redirect-rules.js:423]: is it possible to rewrite these to map to `/ci/` instead of `/nx-cloud/` to reduce the redirects?

Comment (rarmatei) [reply nx-dev/nx-dev/redirect-rules.js:423]: 👍 makes sense, will do

**Review by nartc — APPROVED** [review]
the docs content lgtm

**Review by ndcunningham — APPROVED** [review]
LGTM

**Pull request merged**